File: frontend/lib/models/company.dart
class Company {
  final String id;
  final String company_slug;
  final String cv_id;
  final String cover_letter_id;
  final int order_version;

  Company({
    required this.id,
    required this.company_slug,
    required this.cv_id,
    required this.cover_letter_id,
    required this.order_version,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: json['id'] as String,
      company_slug: json['company_slug'] as String,
      cv_id: json['cv_id'] as String,
      cover_letter_id: json['cover_letter_id'] as String,
      order_version: json['order_version'] as int,
    );
  }
}

File: api/app/Models/Company.php
<?php

/**
 * Created by Reliese Model.
 */

namespace App\Models;

use Illuminate\Database\Eloquent\Collection;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
/**
 * Class Company
 * 
 * @property uuid $id
 * @property string|null $company_slug
 * @property uuid|null $cv_id
 * @property uuid|null $cover_letter_id
 * 
 * @property Cv|null $cv
 * @property CoverLetter|null $cover_letter
 * @property Collection|Notepad[] $notepads
 *
 * @package App\Models
 */
class Company extends Model
{
	use HasFactory, HasUuids;

	protected $table = 'companies';
	public $incrementing = false;
	public $timestamps = false;

	protected $casts = [
		'id' => 'string',
		'cv_id' => 'string',
		'cover_letter_id' => 'string',
		'order_version' => 'int'
	];

	protected $fillable = [
		'company_slug',
		'cv_id',
		'cover_letter_id',
		'order_version'
	];

	public function cv()
	{
		return $this->belongsTo(Cv::class);
	}

	public function cover_letter()
	{
		return $this->belongsTo(CoverLetter::class);
	}

	public function notepads()
	{
		return $this->hasMany(Notepad::class);
	}
}
